kaos/algorithm/interpolator.py

```diff
@@ -30,39 +58,69 @@ def linear_interp(platform_id, timestamp):
         # find the correct segment
         segment = OrbitSegment.get_by_platform_and_time(platform_id, timestamp)
         if not segment:
-            raise ValueError("No segment found: {}, {}".format(platform_id, timestamp))
+            raise InterpolationError("No segment found: {}, {}".format(platform_id, timestamp))
 
         orbit_records = OrbitRecord.get_by_segment(segment.segment_id)
 
         # can't do linear interpolation if we don't have enough records
         if not orbit_records or len(orbit_records) < 2:
-            raise ValueError("Not enough records to perform interpolation: {}, {}".format(
+            raise InterpolationError("Not enough records to perform interpolation: {}, {}".format(
                 platform_id, timestamp))
 
-        # pylint: disable=invalid-name
-        xp = np.array([rec.time for rec in orbit_records])
-        pos = np.array([np.array(rec.position) for rec in orbit_records])
-        vel = np.array([np.array(rec.velocity) for rec in orbit_records])
-
-        tpos = np.zeros(3)
-        tvel = np.zeros(3)
+        times = np.array([rec.time for rec in orbit_records])
+        positions = np.array([np.array(rec.position) for rec in orbit_records])
+        velocities = np.array([np.array(rec.velocity) for rec in orbit_records])
 
-        for i in range(3):
-            tpos[i] = np.interp(x=timestamp, xp=xp, fp=pos[:, i])
-            tvel[i] = np.interp(x=timestamp, xp=xp, fp=vel[:, i])
+        # do the interpolation
+        pos = Interpolator.vector_interp(times, positions, [timestamp], kind="linear")[0]
+        vel = Interpolator.vector_interp(times, velocities, [timestamp], kind="linear")[0]
 
-        return tuple(tpos), tuple(tvel)
+        return tuple(pos), tuple(vel)
 
-    def interpolate(self, timestamp):
+    def interpolate(self, timestamp, kind="quadratic"):
```

Comment:
I dont think you should add this if its not used!

Reply:
We may want to tweak the kind in the future. For example, "cubic" would be more accurate (but more computationally intensive). Having it as a default parameter makes it easy to test -- don't have to modify the source to try out a new setting.